xdsl/dialects/stencil.py

```diff
@@ -406,7 +406,7 @@ class TempType(
     name = "stencil.temp"
 
 
-StencilTypeConstr = StencilType[Attribute].constr()
+StencilTypeConstr = FieldType.constr() | TempType.constr()
```

Comment:
this was unfortunately a blocker for @n-io, I'd be up for saying that we currently don't support this pattern if it's a blocker for this PR, but we should check in first whether this breaks anything. If we do go ahead, I would probably rather first refactor StencilType to be a Union instead of a superclass.

Reply:
I have been meaning to transform StencilType to this union for a while. I think nothing breaks, but it fundamentally makes the machinery less extensible. Now that the paper is submitted I would propose discussing this again. Could be a good topic for Monday's xDSL meeting.

Reply:
I had a go and failed, happy to approve this change.